### ISSUE TITLE
[0.9.5.x] Missing user agent on cache priming

### ISSUE THREAD

**Issue by szepeviktor** [szepeviktor/fix-w3tc#262]
```apache
1.2.3.4 - - [07/Jan/2017:19:07:02 +0000] "GET /sitemap_index.xml HTTP/1.1" 200 622 "http://*/sitemap_index.xml" "W3 Total Cache"
```
The sitemap request has a user agent, but the subsequent requests lack one.

**Comment by szepeviktor** [comment]
I see `'user-agent' => ''` in PgCache_Plugin_Admin.php but an empty useragent keeps the request undisclosed.
https://github.com/szepeviktor/w3-total-cache-fixed/blob/v0.9.5.x/PgCache_Plugin_Admin.php#L147

Could we have something like in the core?

`'user-agent' => apply_filters( 'http_headers_useragent', 'WordPress/' . get_bloginfo( 'version' ) . '; ' . get_bloginfo( 'url' ) ),`

**Comment by nigrosimone** [comment]
mhh w3tc use `w3_http_request` for all http request and it seems ever set the user agent at line [33](https://github.com/szepeviktor/w3-total-cache-fixed/blob/7ec502c177b0f03a5365cf5d7a7c7d6aaf85f8c7/inc/functions/http.php#L33)

**Comment by szepeviktor** [comment]
@nigrosimone It is about 0.9.5.x

**Comment by nigrosimone** [comment]
ah ok... reading the comment into the [code]([here](https://github.com/szepeviktor/w3-total-cache-fixed/blob/v0.9.5.x/PgCache_Plugin_Admin.php#L150)]) it seems a design choise of w3tc developers:
```php
// use empty user-agent since by default we use W3TC-powered by
// which blocks caching
foreach ( $queue as $url )
    Util_Http::get( $url, array( 'user-agent' => '' ) );
```
it's a strange choise because some web firewall block all request without user agent..... my proposal is:
```php
// use WordPress user-agent since by default we use W3TC-powered by
// which blocks caching
foreach ( $queue as $url )
    Util_Http::get( $url, array( 'user-agent' => 'WordPress' ) );
```

**Comment by szepeviktor** [comment]
OK!

**Comment by szepeviktor** [comment]
Please see other occurrences of `'user-agent'`

**Comment by nigrosimone** [comment]
0.9.5.x PR https://github.com/szepeviktor/w3-total-cache-fixed/pull/263
0.9.4.x PR https://github.com/szepeviktor/w3-total-cache-fixed/pull/264

> Please see other occurrences of 'user-agent'

It seems the only one line of code withou a user agent
